pkg/api/validate/dynamic/dynamic.go

```diff
@@ -229,6 +235,55 @@ func (dv *dynamic) validateRouteTablePermissions(ctx context.Context, s Subnet)
 	return err
 }
 
+// validateNatGatewayPermissions will validate permissions on provided subnet
+func (dv *dynamic) validateNatGatewayPermissions(ctx context.Context, s Subnet) error {
+	dv.log.Printf("validateNatGatewayPermissions")
+
+	vnetID, _, err := subnet.Split(s.ID)
+	if err != nil {
+		return err
+	}
+
+	vnetr, err := azure.ParseResourceID(vnetID)
+	if err != nil {
+		return err
+	}
+
+	vnet, err := dv.virtualNetworks.Get(ctx, vnetr.ResourceGroup, vnetr.ResourceName, "")
+	if err != nil {
+		return err
+	}
+
+	ngID, err := getNatGatewayID(&vnet, s.ID)
+	if err != nil || ngID == "" { // error or no nat gateway
+		return err
+	}
```

Comment:
what does that mean if `ngID == ""` ?

Reply:
The data type of ngID is string. In this statement, we need to check if there is no Nat Gateway associated to the Subnet.

Reply:
but does that mean that this is a valid case? 
I would rather have a separate `if` if that is the case. 
 ```suggestion
	ngID, err := getNatGatewayID(&vnet, s.ID)
	if err != nil {
		return err
	}
	if ngID == "" { // no nat gateway so valid
		return nil
	}
```

Reply:
Suggested changes implemented.